### PR TITLE
{bio}[foss/2016b] GlimmerHMM 3.0.4

### PR DESCRIPTION
**Pull request by SethosII**
This adds GlimmerHMM.

It has two directories, `source` and `train`, where source files are located. Currently, the EasyConfig only builds glimmerhmm from the `source` directory and copies the precompiled binaries from the `train` directory to bin. What would be a proper way to also trigger the build in the train directory, `buildopts` with `cd` to and `make` in the train directory?

### CODE REVIEW

**Comment by boegel** [comment]
@SethosII Yes, something like:

```python
# also build in 'train' subdirectory to overwrite pre-compiled binaries
buildopts = " && cd ../train && make "
```

File: easybuild/easyconfigs/g/GlimmerHMM/GlimmerHMM-3.0.4-foss-2016b.eb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+source_urls = ['ftp://ccb.jhu.edu/pub/software/%(namelower)s']
+sources = [SOURCE_TAR_GZ]
+
+checksums = ['43e321792b9f49a3d78154cbe8ddd1fb747774dccb9e5c62fbcc37c6d0650727']
+
+start_dir = 'sources'
+
+files_to_copy = [(['sources/%(namelower)s'], 'bin'), (['train/%s' % x for x in ['build1', 'build2',
+                   'build-icm', 'build-icm-noframe', 'erfapp', 'falsecomp', 'findsites', 'karlin',
```
Comment (boegel) [inline easybuild/easyconfigs/g/GlimmerHMM/GlimmerHMM-3.0.4-foss-2016b.eb]: `'build-icm'` shouldbe aligned with `'build1'` on line above

better approach is probably to use a local variable for the list of things to copy from `train/`:

```python
train_files = ['build1', ... ]
files_to_copy = [(['sources/%(namelower)s'], 'bin'), (['train/%s' % x for x in train_files], 'bin'), 'trained_dir']
```

Are you sure all those `train` files need to be copied though, is that mentioned in the install guide?

Comment (SethosII) [reply easybuild/easyconfigs/g/GlimmerHMM/GlimmerHMM-3.0.4-foss-2016b.eb]: @boegel The readme.train only tells to run `make` in the `train` directory and all those files are created by the makefile. So I would just copy them all to the `bin` directory. The train files are only needed if one wants to train GlimmerHMM differently (already pre-trained, that's what trained_dir contains).

**Comment by SethosII** [comment]
@boegel I added a `buildopts` line.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node2066.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/13caea35f2f0ea9c732efbd2ed989f40 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node2406.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/ee7b2f54aba108a3bfab6e0c57b46670 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2406.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/6d801160b2f577046555b21193dd6bab for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2066.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/91d0e5d847e5467ecec0eb2e84585d05 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @SethosII!

**Pull request merged**